include/aikido/robot/ConcreteRobot.hpp

```diff
@@ -0,0 +1,267 @@
+#ifndef AIKIDO_ROBOT_CONCRETEROBOT_HPP_
+#define AIKIDO_ROBOT_CONCRETEROBOT_HPP_
+
+#include <chrono>
+#include <string>
+#include <unordered_map>
+#include <Eigen/Core>
+#include <dart/dart.hpp>
+#include "aikido/common/ExecutorThread.hpp"
+#include "aikido/common/RNG.hpp"
+#include "aikido/constraint/dart/CollisionFree.hpp"
+#include "aikido/constraint/dart/TSR.hpp"
+#include "aikido/control/TrajectoryExecutor.hpp"
+#include "aikido/planner/parabolic/ParabolicSmoother.hpp"
+#include "aikido/planner/parabolic/ParabolicTimer.hpp"
+#include "aikido/robot/Robot.hpp"
+#include "aikido/robot/util.hpp"
+#include "aikido/statespace/dart/MetaSkeletonStateSpace.hpp"
+#include "aikido/trajectory/Trajectory.hpp"
+
+namespace aikido {
+namespace robot {
+
+AIKIDO_DECLARE_POINTERS(ConcreteRobot)
+
+class ConcreteRobot : public Robot
+{
+public:
+  ConcreteRobot(
+      const std::string& name,
+      dart::dynamics::MetaSkeletonPtr robot,
+      bool simulation,
+      std::unique_ptr<aikido::common::RNG> rng,
+      aikido::control::TrajectoryExecutorPtr trajectoryExecutor,
+      dart::collision::CollisionDetectorPtr collisionDetector,
+      dart::collision::CollisionGroupPtr collideWith,
+      std::shared_ptr<dart::collision::BodyNodeCollisionFilter>
+          selfCollisionFilter);
+
+  virtual ~ConcreteRobot() = default;
+
+  // Documentation inherited.
+  virtual std::unique_ptr<aikido::trajectory::Spline> smoothPath(
+      const dart::dynamics::MetaSkeletonPtr& metaSkeleton,
+      const aikido::trajectory::Trajectory* path,
+      const constraint::TestablePtr& constraint) override;
+
+  // Documentation inherited.
+  virtual std::unique_ptr<aikido::trajectory::Spline> retimePath(
+      const dart::dynamics::MetaSkeletonPtr& metaSkeleton,
+      const aikido::trajectory::Trajectory* path) override;
+
+  // Documentation inherited.
+  virtual std::future<void> executeTrajectory(
+      const trajectory::TrajectoryPtr& trajectory) override;
+
+  // Documentation inherited.
+  virtual boost::optional<Eigen::VectorXd> getNamedConfiguration(
+      const std::string& name) const override;
+
+  // Documentation inherited.
+  virtual void setNamedConfigurations(
+      std::unordered_map<std::string, const Eigen::VectorXd>
+          namedConfigurations) override;
+
+  // Documentation inherited.
+  virtual std::string getName() const override;
+
+  // Documentation inherited.
+  virtual dart::dynamics::MetaSkeletonPtr getMetaSkeleton() override;
```

Comment:
nit: function should be declared `const`?

Reply:
I'm keeping this as per @jslee02 's comment above. Additionally, we can provide a const version of this which returns ConstMetaSkeletonPtr but I'm not sure whether that's useful.